### PR TITLE
Switch to NpmClientHandler

### PR DESCRIPTION
**Pull request by rchl**

### CODE REVIEW

File: plugin.py [hunk 1]
```diff
@@ -1,101 +1,44 @@
...
-        return server.ready
+        configuration['settings'].setdefault('json', {})['schemas'] = self._default_schemas
 
-    def on_initialized(self, client) -> None:
-        client.on_request(
+    def on_ready(self, api) -> None:
+        api.on_request(
             'vscode/content',
-            lambda params, request_id: self.handle_vscode_content(client, params, request_id))
+            lambda params, request_id: self.handle_vscode_content(params, request_id))
```
Comment (predragnikolic) [inline plugin.py:39]:
```
api.on_request(
            'vscode/content',
            self.handle_vscode_content)
```
Shouldn't this also also work? :)

File: plugin.py [hunk 2]
```diff
@@ -1,101 +1,44 @@
...
 
-    def handle_vscode_content(self, client, uri, request_id):
+    def handle_vscode_content(self, uri, respond):
```
Comment (predragnikolic) [inline plugin.py:41]: I was a bit confused at first glance on how the `request_id`(a string) became a `respond` function.

Found the answer at [`ApiWrapper`](https://github.com/sublimelsp/lsp_utils/blob/6fc936b7991a72b66f78a26bf7bde88e80288bcd/st3/lsp_utils/npm_client_handler.py#L31-L32).

File: plugin.py (hunk 1 above)
Comment (rchl) [reply plugin.py:39]: Yes. :)
And that was also confusing as argument names were wrong :)

File: plugin.py (hunk 2 above)
Comment (rchl) [reply plugin.py:39]: Well that was just my bad with naming. It worked but names were all mixed up here. Now fixed. :)

**Pull request merged**